src/hotspot/share/opto/graphKit.cpp

```diff
@@ -3469,6 +3469,8 @@ FastLockNode* GraphKit::shared_lock(Node* obj) {
 
   // Box the stack location
   Node* box = _gvn.transform(new BoxLockNode(next_monitor()));
+  // Check for bailout after new BoxLockNode
+  if (failing()) { return nullptr; }
```

Comment:
Do all callers of `shared_lock()` checks for `failing()` or returned `nullptr`?

Reply:
No, not the immediate callers at least. Below is a quick call graph analysis for the places where we create `BoxLockNode`s (up to the first bailout check). Should I add returns at all points in the call chain up to the first checks?
```
graphKit.cpp:3471 (this is in shared_lock)
  locknode.cpp:196
    parse2.cpp:2759
      parse1.cpp:1594 (Checks for bailout at parse1.cpp:1595)
  parse1.cpp:1264
    parse1.cpp:582 (Checks for bailout at parse1.cpp:596)

parse1.cpp:227
  parse1.cpp:579 (Checks for bailout at parse1.cpp:596)
```

Reply:
I am not worry about exit in`load_interpreter_state()` there is check after call.

There is call to `record_profiled_parameters_for_speculation()` after `shared_lock()` call in `do_method_entry()`. It does not check `failing()`. May be add check `if (!failing())` when calling it (or inside it).

For `do_one_bytecode()` case the only question is if IGVN code can handle such bailout. It is called at the end of method. If not, we need to add the check.

Reply:
> There is call to record_profiled_parameters_for_speculation() after shared_lock() call in do_method_entry(). It does not check failing(). May be add check if (!failing()) when calling it (or inside it).

I've now added a check just after `_synch_lock = shared_lock(lock_obj);`, thanks.

> For do_one_bytecode() case the only question is if IGVN code can handle such bailout. **It is called at the end of method**. If not, we need to add the check.

I cannot see any call to IGVN after `do_monitor_enter()` in `do_one_bytecode()`. Can you elaborate? Maybe related, I also changed to
```
Node *box = new BoxLockNode(next_monitor());
// Check for bailout after new BoxLockNode
if (failing()) { return; }
box = _gvn.transform(box);
```
from the previous
```
Node *box = _gvn.transform(new BoxLockNode(next_monitor()));
// Check for bailout after new BoxLockNode
if (failing()) { return; }
```
to avoid potential issues with `_gvn.transform` due to the new bailout.

Reply:
Does `_gvn.transform` handle bailout?

> I cannot see any call to IGVN after `do_monitor_enter()` in `do_one_bytecode()`. Can you elaborate?
Sorry, I mean IGV (IdealGraphVisualizer) [parse2.cpp#L2782](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/parse2.cpp#L2782)